src/catalog/catalog.cpp

```diff
@@ -357,7 +389,8 @@ storage::Database *Catalog::GetDatabaseWithOid(const oid_t db_oid) const {
 storage::Database *Catalog::GetDatabaseWithName(const std::string database_name)
     const {
   for (auto database : databases_) {
-    if (database->GetDBName() == database_name) return database;
+    if (database != nullptr && database->GetDBName() == database_name)
```

Comment:
This usually doesn't happen.. It's todo with catalog destruction. Previously I created catalog_db and default_db on catalog bootstrap, upon destruction if catalog_db is deleted first, then GetDatabaseWithName will segfault when looking up default_db.